### DISCUSSION TITLE
Ability to clear a specific DAG Run's task instances via REST APIs

### DISCUSSION THREAD

**Discussion by yashk97** [apache/airflow#23220] [General] [answered]
Hi,

My use case is in case multiple DAG Runs fail on some task (not the same one in all of them), I want to individually re-trigger each of these DAG Runs. Currently, I have to rely on the Airflow UI (attached screenshots) where I select the failed task and clear its state (along with the downstream tasks) to re-run from that point. While this works, it becomes tedious if the number of failed DAG runs is huge.

I checked the REST API Documentation and came across the clear Task Instances API with the following URL: /api/v1/dags/{dag_id}/clearTaskInstances

However, it filters task instances of the specified DAG in a given date range. 

I was wondering if, for a specified DAG Run, we can clear a task along with its downstream tasks irrespective of the states of the tasks or the DAG run through REST API.

This will give us more granular control over re-running DAGs from the point of failure.
![image](https://user-images.githubusercontent.com/25115516/165099593-46ce449a-d303-49ee-9edb-fc5d524f4517.png)
![image](https://user-images.githubusercontent.com/25115516/165099683-4ba7f438-3660-4a16-a66c-2017aee5042f.png)

**Comment by bbovenzi** [comment, marked as answer]
Yes, we should move that endpoint (and many others) from the webserver to the REST API. I like passing a range of dates but a specific dag_run_id should also be allowed.

I think this is better as a feature request. https://github.com/apache/airflow/issues/23227
